Content.Shared/Silicons/StationAi/SharedStationAiSystem.Held.cs

```diff
@@ -120,29 +122,42 @@ private void OnHeldInteraction(Entity<StationAiHeldComponent> ent, ref Interacti
         // Cancel if it's not us or something with a whitelist.
         args.Cancelled = ent.Owner != args.Target &&
                          args.Target != null &&
-                         (!TryComp(args.Target, out StationAiWhitelistComponent? whitelist) || !whitelist.Enabled);
+                         !HasComp<StationAiWhitelistComponent>(args.Target);
     }
 
     private void OnTargetVerbs(Entity<StationAiWhitelistComponent> ent, ref GetVerbsEvent<AlternativeVerb> args)
     {
-        if (!args.CanComplexInteract ||
-            !ent.Comp.Enabled ||
-            !HasComp<StationAiHeldComponent>(args.User) ||
-            !HasComp<StationAiWhitelistComponent>(args.Target))
+        if (!args.CanComplexInteract
+            || !HasComp<StationAiHeldComponent>(args.User)
+            || !HasComp<StationAiWhitelistComponent>(args.Target))
         {
             return;
         }
 
         var user = args.User;
+        if (!ent.Comp.Enabled)
+        {
+            _popup.PopupClient(Loc.GetString("ai-device-not-responding"), user, PopupType.MediumCaution);
+            return;
+        }
+
         var target = args.Target;
 
         var isOpen = _uiSystem.IsUiOpen(target, AiUi.Key, user);
 
-        args.Verbs.Add(new AlternativeVerb()
+        var verb = new AlternativeVerb
         {
             Text = isOpen ? Loc.GetString("ai-close") : Loc.GetString("ai-open"),
             Act = () =>
             {
+                // no need to show menu if device is not powered.
+                SharedApcPowerReceiverComponent? component = null;
+                if (PowerReceiver.ResolveApc(ent.Owner, ref component) && !component.Powered)
+                {
+                    _popup.PopupClient(Loc.GetString("ai-device-not-responding"), user, PopupType.MediumCaution);
+                    return;
+                }
+
```

Comment:
Please do not hardcode in devices here, it's supposed to be generic. Use an event or anything else.

Reply:
I swear PowerReceiver.IsPowered was throwing exception... but now i tested it and it works fine for all cases. Oh well...

Reply:
fixed